tools/topology/topology2/include/pipelines/cavs/src-gain-playback.conf

```diff
@@ -0,0 +1,186 @@
+			num_audio_formats 6
+			# 16-bit 48KHz 2ch
+			Object.Base.audio_format.1 {
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$ibs * 2]"
+			}
+			Object.Base.audio_format.2 {
+				in_bit_depth		32
+				in_valid_bit_depth	24
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$obs * 2]"
+			}
+			Object.Base.audio_format.3 {
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$ibs * 2]"
+			}
+
+			# 44.1khz input
+			Object.Base.audio_format.4 {
+				in_rate                 44100
+				in_bit_depth            16
+				in_valid_bit_depth      16
+				out_bit_depth           32
+				out_valid_bit_depth     32
+				dma_buffer_size "$[$obs * 2]"
+			}
+
+			Object.Base.audio_format.5 {
+				in_rate                 44100
+				in_bit_depth            32
+				in_valid_bit_depth      24
+				out_bit_depth           32
+				out_valid_bit_depth     32
+				dma_buffer_size "$[$obs * 2]"
+			}
+
+			Object.Base.audio_format.6{
+				in_rate                 44100
+				in_bit_depth            32
+				in_valid_bit_depth      32
+				out_bit_depth           32
+				out_valid_bit_depth     32
+				dma_buffer_size "$[$obs * 2]"
+			}
+
+			node_type $HDA_HOST_OUTPUT_CLASS
+		}
+
+		src."1" {
+			num_audio_formats 5
+
+			Object.Base.audio_format.1 {}
+
+			Object.Base.audio_format.2 {
+				in_bit_depth            32
+				in_valid_bit_depth      32
+				out_bit_depth           32
+				out_valid_bit_depth     32
+			}
+
+			Object.Base.audio_format.3 {
+				in_rate                44100
+				in_bit_depth           16
+				in_valid_bit_depth     16
+				out_bit_depth          16
+				out_valid_bit_depth    16
+				# src needs extra 4 sample size
+				obs "$[($out_channels * $[($[($out_rate / 1000)] + 4)]) * ($out_bit_depth / 8)]"
+			}
+
+			Object.Base.audio_format.4 {
+				in_rate                 44100
+				in_bit_depth            32
+				in_valid_bit_depth      24
+				out_bit_depth           32
+				out_valid_bit_depth     24
+				obs "$[($out_channels * $[($[($out_rate / 1000)] + 4)]) * ($out_bit_depth / 8)]"
+			}
+
+			Object.Base.audio_format.5 {
+				in_rate                 44100
+				in_bit_depth            32
+				in_valid_bit_depth      32
+				out_bit_depth           32
+				out_valid_bit_depth     32
+				obs "$[($out_channels * $[($[($out_rate / 1000)] + 4)]) * ($out_bit_depth / 8)]"
+			}
+		}
+
+		gain."1" {
+			num_audio_formats 2
+			#16-bit 48KHz 2ch
+			Object.Base.audio_format.1 {}
+			# 32-bit 48KHz 2ch
+			Object.Base.audio_format.2 {
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_bit_depth		32
+				out_valid_bit_depth	32
+			}
+		}
+
+		pipeline."1" {
+			priority	0
+			lp_mode		0
+		}
+	}
+
+	Object.Base {
+		route."1" {
+			source	"copier.host..1"
+			sink	"src..1"
+		}
+
+		route."2" {
+			source  "src..1"
+			sink    "gain..1"
+		}
+	}
+
+	direction	"playback"
+	dynamic_pipeline 1
+	time_domain	"timer"
```

Comment:
**Good question. It is cpc which is stored in module config in fw manifest, please check the CPC in the following table.  Driver should match the entry by pcm parameters and get CPC.  The total CPC value in all module will be added and transformed to cpc for pipeline creation (Maybe drive doesn't support it now). The close source fw will use this cpc to schedule the pipeline. Currently it is not used by SOF IPC3 & IPC4. May be it will be supported by IPC4 in future.**


````
	# mod_cfg [PAR_0 PAR_1 PAR_2 PAR_3 IS_BYTES CPS IBS OBS MOD_FLAGS CPC OBLS]
	mod_cfg = [0,    0,     0,    0,     12832,  1365500, 0, 0, 0,   1365, 0,
		   1, 0, 0, 0, 12832, 2302300, 0, 0, 0, 2302, 0,
		2, 0, 0, 0, 12832, 3218200, 0, 0, 0, 3218, 0,
		3, 0, 0, 0, 12832, 4169700, 0, 0, 0, 4169, 0,
		4, 0, 0, 0, 12832, 5095100, 0, 0, 0, 5095, 0,
		5, 0, 0, 0, 12832, 6014800, 0, 0, 0, 6014, 0,
		6, 0, 0, 0, 12832, 6963500, 0, 0, 0, 6963, 0,
		7, 0, 0, 0, 12832, 7791000, 0, 0, 0, 7791, 0,
		8, 0, 0, 0, 12832, 8843000, 0, 0, 0, 8843, 0,
		9, 0, 0, 0, 12832, 9755100, 0, 0, 0, 9755, 0,
		10, 0, 0, 0, 12832, 10726500, 0, 0, 0, 10726, 0,
		11, 0, 0, 0, 12832, 11624100, 0, 0, 0, 11624, 0,
		12, 0, 0, 0, 12832, 12518700, 0, 0, 0, 12518, 0,
		13, 0, 0, 0, 12832, 13555000, 0, 0, 0, 13555, 0,
		14, 0, 0, 0, 12832, 14144500, 0, 0, 0, 14144, 0,
		15, 0, 0, 0, 12832, 15809800, 0, 0, 0, 15809, 0,
		16, 0, 0, 0, 12832, 16749000, 0, 0, 0, 16749, 0,
		17, 0, 0, 0, 12832, 18433500, 0, 0, 0, 18433, 0,
		18, 0, 0, 0, 12832, 19425900, 0, 0, 0, 19425, 0,
		19, 0, 0, 0, 12832, 20396900, 0, 0, 0, 20396, 0,
		20, 0, 0, 0, 12832, 20881000, 0, 0, 0, 20881, 0,
		21, 0, 0, 0, 12832, 23431000, 0, 0, 0, 23431, 0,
		22, 0, 0, 0, 12832, 30471000, 0, 0, 0, 30471, 0]
````

Reply:
How to match it ? it is a puzzle to us. Driver uses a complex algorithm and we need time to decode it. I can't manage to understand it at first glance. Do we need to support it ? As these numbers are built with close source fw so they are not useful for sof fw in future.

Reply:
Some of these values are used by the Windows driver. There may be future schedulers that could use some of this data, but tooling to generate this data is also needed. Currently not needed so can be set to 0 for compatibility.